avm/res/kusto/cluster/database/principal-assignment/main.bicep

```
@@ -0,0 +1,61 @@
+metadata name = 'Kusto Cluster Database Principal Assignments'
+metadata description = 'This module deploys a Kusto Cluster Database Principal Assignment.'
+
+@minLength(4)
+@maxLength(22)
+@description('Conditional. The name of the parent Kusto Cluster Database. Required if the template is used in a standalone deployment.')
+param kustoClusterDatabaseName string
+
+@description('Required. The principal assignement for the Kusto Cluster Database.')
+param databasePrincipalAssignment databasePrincipalAssignmentType
+
+// ============== //
+// Resources      //
+// ============== //
+
+resource kustoClusterDatabase 'Microsoft.Kusto/clusters/databases@2024-04-13' existing = {
+  name: kustoClusterDatabaseName
+}
+
+resource kustoDatabasePrincipalAssignment 'Microsoft.Kusto/clusters/databases/principalAssignments@2024-04-13' = {
+  name: databasePrincipalAssignment.principalId
+  parent: kustoClusterDatabase
+  properties: {
+    principalId: databasePrincipalAssignment.principalId
+    principalType: databasePrincipalAssignment.principalType
+    role: databasePrincipalAssignment.role
+    tenantId: databasePrincipalAssignment.tenantId
+  }
+}
```

Here you're missing one level of nesting.

```suggestion
@minLength(4)
@maxLength(22)
@description('Conditional. The name of the Kusto cluster.')
param clusterName string

@minLength(4)
@maxLength(22)
@description('Conditional. The name of the parent Kusto Cluster Database. Required if the template is used in a standalone deployment.')
param databaseName string

resource kustoCluster 'Microsoft.Kusto/clusters@2024-04-13' existing  = {
  name: clusterName

  resource database 'databases@2024-04-13' existing = {
    name: databaseName
  }
}

resource kustoDatabasePrincipalAssignment 'Microsoft.Kusto/clusters/databases/principalAssignments@2024-04-13' = {
  name: databasePrincipalAssignment.principalId
  parent: kustoCluster::databaseName
  properties: {
    principalId: databasePrincipalAssignment.principalId
    principalType: databasePrincipalAssignment.principalType
    role: databasePrincipalAssignment.role
    tenantId: databasePrincipalAssignment.tenantId
  }
}
```

Hence leading to an incorrect segment length (the name has 3 parts, 'cluster/database/assignment', not just 2).

---

Thank you! @AlexanderSehr!